src/query/frontend/ast/cypher_main_visitor.cpp

```diff
@@ -316,6 +316,30 @@ antlrcpp::Any CypherMainVisitor::visitEdgeImportModeQuery(MemgraphCypher::EdgeIm
 antlrcpp::Any CypherMainVisitor::visitSetReplicationRole(MemgraphCypher::SetReplicationRoleContext *ctx) {
   auto *replication_query = storage_->Create<ReplicationQuery>();
   replication_query->action_ = ReplicationQuery::Action::SET_REPLICATION_ROLE;
+
+#ifdef MG_ENTERPRISE
+  // Licence check in the interpreter.
+  if (ctx->MAIN() || ctx->REPLICA()) {
+    if (!ctx->WITH() || !ctx->PORT()) {
+      throw SemanticException("Port must be specified when setting replication role to main or replica!");
```

Comment:
I think the port should be optional. We might be enterprise and still not use the coordinator. The port is superfluous in that case.